Test/civl/inductive-sequentialization/Gauss.bpl

```diff
@@ -0,0 +1,58 @@
+// RUN: %boogie "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+var {:layer 0,2} x:int;
+
+type {:pending_async}{:datatype} PA;
+function {:constructor} ADD(i: int) : PA;
+
+////////////////////////////////////////////////////////////////////////////////
+
+function trigger(i: int) : bool { true }
+
+procedure {:atomic}{:layer 1}
+{:IS "MAIN'","INV"}{:elim "ADD"}
+SUM (n: int)
+returns ({:pending_async "ADD"} PAs:[PA]int)
+modifies x;
+{
+  assert n >= 0;
+  assert trigger(0); // base hint
+  PAs := (lambda pa: PA :: if is#ADD(pa) && 1 <= i#ADD(pa) && i#ADD(pa) <= n then 1 else 0);
+}
+
+procedure {:atomic}{:layer 2}
+MAIN' (n: int)
+modifies x;
+{
+  assert n >= 0;
+  x := x + ((n * (n+1)) div 2);
```

Comment:
you might not need one level of parentheses.

Reply:
Right, I updated it. It would even be possible to drop the parentheses around `n * (n+1)`, but this is getting too close to anarchy for me :smile: